hw/top_earlgrey/data/clocks.xdc

```diff
@@ -72,39 +90,166 @@ set_input_delay  -add_delay -clock_fall -clock jtag_tck -min  0.0 [get_ports {IO
 set_input_delay  -add_delay -clock_fall -clock jtag_tck -max 15.0 [get_ports {IOR0 IOR2}]
 
 ## SPI clocks
-create_clock -add -name clk_spi  -period 100.00 -waveform {0 50} [get_ports SPI_DEV_CLK]
-set_input_delay -clock clk_spi 5 [get_ports SPI_DEV_D0]
-set_input_delay -clock clk_spi 5 [get_ports SPI_DEV_D1]
-set_output_delay -clock clk_spi 5 [get_ports SPI_DEV_D0] -add_delay
-set_output_delay -clock clk_spi 5 [get_ports SPI_DEV_D1] -add_delay
-
-# set clock sense on the input to spi buffers to help the tool understand the clocks are shifted versions of each other
-# This can also be accomplished through create_genearted_clocks.
-
-## set clock sense approach
-##set_clock_sense -negative \
-##  [get_pins -filter {DIRECTION == OUT && IS_LEAF} -of_objects \
-##    [get_nets -segments -of_objects \
-##      [get_pins top_*/u_spi_device/gen_fpga_buf.gen_bufr.bufr_i_i_1/I] \
-##    ] \
-##  ] \
-##  -clocks clk_spi
-##
-##set_clock_sense -positive \
-##  [get_pins -filter {DIRECTION == OUT && IS_LEAF} -of_objects \
-##    [get_nets -segments -of_objects \
-##      [get_pins top_*/u_spi_device/gen_fpga_buf.gen_bufr.bufr_i_i_1__0/I] \
-##    ] \
-##  ] \
-##  -clocks clk_spi
-
-## create_generated_clock appraoch
-## create_generated_clock is preferred since the buffer cell used here is hand-instantiated, while the set_clock_sense point is simply a LUT
+set spi_dev_period 100.0
+set spi_dev_half_period [expr ${spi_dev_period} / 2]
+# Max board skew between signals
+set spi_dev_board_skew  0.5
+# Max board delay
+set spi_dev_board_delay 0.6
+# Board skew affects input path for sampling
+set spi_dev_in_delay_min [expr -2.0 - ${spi_dev_board_skew}]
+set spi_dev_in_delay_max [expr  3.0 + ${spi_dev_board_skew}]
+# The board delay affects time remaining on the output path.
+set spi_dev_out_hold      -5.0
+set spi_dev_out_setup     [expr  5.0 + 2 * ${spi_dev_board_delay}]
+
+create_clock -add -name clk_spi  -period ${spi_dev_period} \
+    -waveform "0 ${spi_dev_half_period}" [get_ports SPI_DEV_CLK]
+# CSB must act as a clock, in addition to data and a reset.
```

Comment:
do we need to constrain the `cio_tpm_csb_i` input separately, or do you think the wildcard constraint for pinmux paths to  IO_DIV2/4 should be sufficient for that?

Reply:
Due to the asynchronous clock groups, a wildcard constraint for IO_DIV2/4 won't be part of the timing for SPI_DEV_CLK. We'll need something specific for that.

Reply:
Yeah I initially thought the constraints wrt to the IO_DIV2/4 clocks would at least make sure that there is some delay constraint in place - but I see now that this also feeds into a reset and has combo output paths like this one:
https://github.com/lowRISC/opentitan/blob/acd3be0765f70d31b531d9dbdee1a2a5fd6c0123/hw/ip/spi_device/rtl/spi_device.sv#L1298

So yeah it looks like we need something separate for that.